crates/stackable-telemetry/src/tracing/settings/mod.rs

```diff
@@ -105,6 +108,11 @@ impl SettingsBuilder {
         self.into()
     }
 
+    /// Set specific [`FileLogSettings`].
+    pub fn file_log_settings_builder(self) -> FileLogSettingsBuilder<builder_state::PreLogDir> {
```

Comment:
note: I think it makes sense to require the user to pass the folder path here as it is required. This then also should get rid of the builder states.

The `with_` functions are intended for optional settings (which override some kind of default value).